### ISSUE TITLE
AudioStreamPlayer not paused on pause mode change

### ISSUE THREAD

**Issue by KoBeWi** [godotengine/godot#58543]
### Godot version

3.x 0959ff2

### System information

W10

### Issue description

When you have AudioStreamPlayer and change its `pause_mode`, it will not get paused when it should.

Example code:
(`$AudioStreamPlayer` has `PAUSE_MODE_PROCESS`)
```
func _ready() -> void:
	get_tree().paused = true
	$AudioStreamPlayer.play()
	yield(get_tree().create_timer(0.5), "timeout")
	$AudioStreamPlayer.pause_mode = Node.PAUSE_MODE_STOP # <- should stop playing here
```
Does not happen on master.

### Steps to reproduce

1. Add AudioStreamPlayer
2. Add any long enough sound
3. Set `pause_mode` to `Process`
4. Add a script with the above code
5. Run and observe the player not pausing

**Comment by pfertyk** [comment]
I'd like to submit a solution. It seems that, even though we save the new `pause_mode`, we don't send any notifications about it. I also added a check for `can_process` when unpausing `AudioStreamPlayer` (similar as pausing).

I tried to keep the code changes to minimum. This solution works for both cases (process => stop and stop => process), but I haven't check all combinations :) It's possible that we could use a new method here, like `propagate_pause_mode`. If so, let me know :)
